mclocations/minecraft.py

```diff
@@ -0,0 +1,89 @@
+import re
+
+from mcrcon import MCRcon
+
+from mclocations.models import Player, Position
+
+
+class MinecraftServer:
+    def __init__(self, host, password, port=25575):
+        self.mcr = MCRcon(host, password, port)
+        self.mcr.connect()
+
+    def disconnect(self):
+        self.mcr.disconnect()
+
+    def get_players(self):
+        players = []
+        player_names = MinecraftServer._parse_list(self.mcr.command('/list'))
+
+        for player_name in player_names:
+            position = MinecraftServer._parse_position(self.mcr.command(f'/data get entity {player_name} Pos'))
+            dimension = MinecraftServer._parse_dimension(self.mcr.command(f'/data get entity {player_name} Dimension'))
+            players.append(Player(player_name, position, dimension))
+
+        return players
+
+    @classmethod
+    def _parse_list(cls, message):
+        """
+        >>> MinecraftServer._parse_list("There are 0 of a max of 20 players online:")
+        []
+        >>> MinecraftServer._parse_list("There are 1 of a max of 20 players online: _DavidCoe_")
+        ['_DavidCoe_']
+        >>> MinecraftServer._parse_list("There are 2 of a max of 20 players online: _DavidCoe_, _NotDavidCoe_")
+        ['_DavidCoe_', '_NotDavidCoe_']
+
+        :param message:
+        :return:
+        """
+        match = re.match(
+            r'There are (?P<current_count>\d+) of a max of (?P<max_count>\d+) players online:(?: (?P<players>.+))',
+            message)
+        players = []
+
+        if match:
+            players = [p.strip() for p in match.group('players').split(',')]
+
+        return players
+
+    @classmethod
+    def _parse_position(cls, message):
+        """

```

Comment:
@ArchmageInc Add this one to your reading list too. This is the documentation AND the tests!

https://docs.python.org/3/library/doctest.html